### PR TITLE
Add guard

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Have Guard PDB + deployment

@deads2k @soltysh

### CODE REVIEW

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -176,6 +176,10 @@ func createTargetConfigController(ctx context.Context, syncCtx factory.SyncConte
 		errors = append(errors, fmt.Errorf("%q: %v", "configmap", err))
 	}
 	_, _, err = manageClusterPolicyControllerConfig(ctx, c.kubeClient.CoreV1(), syncCtx.Recorder(), operatorSpec)
+	if err != nil {
```
Comment (ravisantoshgudimetla) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:179]: Note to myself: Remove this commit

File: bindata/v4.1.0/kube-controller-manager/guard-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          operator: Exists
+        - key: node.kubernetes.io/unreachable
+          effect: NoExecute
+          operator: Exists
+        - key: node-role.kubernetes.io/etcd
+          operator: Exists
+          effect: NoSchedule
+      containers:
+        - name: guard
+          image: quay.io/openshift/origin-cli:latest
```
Comment (soltysh) [inline bindata/v4.1.0/kube-controller-manager/guard-deployment.yaml:52]: `${CLI_IMAGE}`?

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -74,6 +77,14 @@ func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error
 		cc.EventRecorder,
 	)
 
+	// Add Deployment, PDB Informer for the `openshift-etcd` namespace
```
Comment (soltysh) [inline pkg/operator/starter.go:80]: openshift-etcd?

File: pkg/operator/starter.go [hunk 4]
```diff
@@ -206,6 +221,41 @@ func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error
 	return nil
 }
 
+// withDeploymentChecker checks if the deployment exists or not.
+func withDeploymentChecker(deploymentLister v1.DeploymentLister) resourceapply.ConditionalFunction {
+	return func() bool {
```
Comment (soltysh) [inline pkg/operator/starter.go:226]: Why can't we have the SNO check be part of this check?

**Pull request closed without merging**